nestkernel/conn_builder_conngen.cpp

```diff
@@ -20,158 +20,140 @@
  *
  */
 
-#include "conngen.h"
+#include "conn_builder_conngen.h"
 
-// Includes from libnestutil:
-#include "logging.h"
+#ifdef HAVE_LIBNEUROSIM
 
 // Includes from nestkernel:
-#include "exceptions.h"
 #include "kernel_manager.h"
-#include "modelrange.h"
 
 // Includes from sli:
-#include "token.h"
+#include "dictutils.h"
 
 namespace nest
 {
 
-/**
- * Low-level function of the ConnectionGenerator interface for
- * connecting populations of neurons using a connection generator with
- * a value set and custom synapse type.
- *
- * \param cg The ConnectionGenerator describing the connectivity
- * \param source_node_ids A NodeCollection specifying the source population
- * \param target_node_ids A NodeCollection specifying the target population
- * \param params_map A Dictionary mapping the labels "weight" and
- *        "delay" to their indices in the value set
- * \param synmodel_name The name of the synapse model to use for the
- *        connections
- */
-void
-cg_connect( ConnectionGeneratorDatum& cg,
-  const NodeCollectionPTR source_node_ids,
-  const NodeCollectionPTR target_node_ids,
-  const DictionaryDatum& params_map,
-  const Name& synmodel_name )
+ConnectionGeneratorBuilder::ConnectionGeneratorBuilder( NodeCollectionPTR sources,
+  NodeCollectionPTR targets,
+  const DictionaryDatum& conn_spec,
+  const std::vector< DictionaryDatum >& syn_specs )
+  : ConnBuilder( sources, targets, conn_spec, syn_specs )
+  , cg_( ConnectionGeneratorDatum() )
+  , params_map_()
 {
-  const Token synmodel = kernel().model_manager.get_synapsedict()->lookup( synmodel_name );
-  if ( synmodel.empty() )
+  updateValue< ConnectionGeneratorDatum >( conn_spec, "cg", cg_ );
+  if ( cg_->arity() != 0 )
   {
-    throw UnknownSynapseType( synmodel_name.toString() );
+    if ( not conn_spec->known( "params_map" ) )
+    {
+      throw BadProperty( "A params_map has to be given if the ConnectionGenerator has values." );
+    }
+
+    updateValue< DictionaryDatum >( conn_spec, "params_map", params_map_ );
+
+    for ( Dictionary::iterator it = params_map_->begin(); it != params_map_->end(); ++it )
+    {
+      it->second.set_access_flag();
+    }
+
+    if ( syn_specs[ 0 ]->known( names::weight ) or syn_specs[ 0 ]->known( names::delay ) )
+    {
+      throw BadProperty( "Properties weight and delay cannot be specified in syn_spec if the "
+			 "ConnectionGenerator has values." );
```

Comment:
Is the indentation right here? It looks a bit weird, but it might just be me :)

Reply:
Fixed in c2b4eade0.